@here/harp-mapview/lib/VisibleTileSet.ts

```diff
@@ -597,39 +592,16 @@ export class VisibleTileSet {
             const tiles = renderListEntry.visibleTiles;
             tiles.forEach(tile => {
                 minElevation = Math.min(minElevation, tile.minElevation);
-                maxElevation = Math.max(maxElevation, tile.maxElevation);
+                maxElevation = Math.max(maxElevation, tile.maxElevation + tile.maxGeometryHeight);
             });
         });
-        // We process buildings elevation yet (via ElevationRangeSource or either way), so correct
-        // the highest elevation to account for highest possible building.
-        const projectionScale = this.m_frustumIntersection.projection.getScaleFactor(
-            this.m_frustumIntersection.camera.position
-        );
-        const maxBuildingHeight = EarthConstants.MAX_BUILDING_HEIGHT * projectionScale;
-        maxElevation =
-            maxElevation < EarthConstants.MAX_ELEVATION - maxBuildingHeight
-                ? maxElevation + maxBuildingHeight
-                : maxElevation;
-        // Update elevation ranges stored.
-        if (
-            minElevation !== this.m_elevationRange.minElevation ||
-            maxElevation !== this.m_elevationRange.maxElevation
-        ) {
-            this.m_elevationRange.minElevation = minElevation;
-            this.m_elevationRange.maxElevation = maxElevation;
-        }
-        this.m_elevationRange.calculationStatus = visibleTileKeysResult.allBoundingBoxesFinal
-            ? CalculationStatus.FinalPrecise
-            : CalculationStatus.PendingApproximate;
 
         // If clip planes evaluator depends on the tiles elevation re-calculate
         // frustum planes and update the camera near/far plane distances.
         let viewRangesChanged: boolean = false;
-        if (elevationRangeSource !== undefined) {
-            const viewRanges = this.updateClipPlanes(maxElevation, minElevation);
-            viewRangesChanged = viewRanges === this.m_viewRange;
-            this.m_viewRange = viewRanges;
-        }
+        const viewRanges = this.updateClipPlanes(maxElevation, minElevation);
```

Comment:
Would be nice to add at least some comments that elevation caching is possible, such as together with camera movement information we could possible omit update... But that's later story and I am not sure if worth entire effort.